pandas/core/arrays/interval.py

```diff
@@ -0,0 +1,856 @@
+                                verify_integrity=False)
+
+    # TODO: doc
+    def copy(self, deep=False):
+        left = self.left.copy(deep=True) if deep else self.left
+        right = self.right.copy(deep=True) if deep else self.right
+        closed = self.closed
+        # TODO: Could skip verify_integrity here.
+        return type(self).from_arrays(left, right, closed=closed)
+
+    def _formatting_values(self):
+        return self.values
+
+    def get_values(self):
+        return self.values
+
+    def isna(self):
+        return isna(self.left)
+
+    @property
+    def nbytes(self):
+        return self.left.nbytes + self.right.nbytes
+
+    @property
+    def itemsize(self):
+        return self.left.itemsize + self.right.itemsize
+
+    def take(self, indices, axis=0, allow_fill=True, fill_value=None,
+             **kwargs):
+        nv.validate_take(tuple(), kwargs)
+        indices = _ensure_platform_int(indices)
+        left, right = self.left, self.right
+
+        if fill_value is None:
+            fill_value = self._na_value
+        mask = indices == -1
+
+        if not mask.any():
+            # we won't change dtype here in this case
+            # if we don't need
+            allow_fill = False
+
+        taker = lambda x: x.take(indices, allow_fill=allow_fill,
+                                 fill_value=fill_value)
+
+        try:
+            new_left = taker(left)
+            new_right = taker(right)
+        except ValueError:
+
+            # we need to coerce; migth have NA's in an
+            # integer dtype
+            new_left = taker(left.astype(float))
+            new_right = taker(right.astype(float))
+
+        return self._shallow_copy(new_left, new_right)
+
+    take_nd = take
+
+    # Formatting
+
+    def _format_data(self):
+
+        # TODO: integrate with categorical and make generic
+        # name argument is unused here; just for compat with base / categorical
+        n = len(self)
+        max_seq_items = min((get_option(
+            'display.max_seq_items') or n) // 10, 10)
+
+        formatter = str
+
+        if n == 0:
+            summary = '[]'
+        elif n == 1:
+            first = formatter(self[0])
+            summary = '[{first}]'.format(first=first)
+        elif n == 2:
+            first = formatter(self[0])
+            last = formatter(self[-1])
+            summary = '[{first}, {last}]'.format(first=first, last=last)
+        else:
+
+            if n > max_seq_items:
+                n = min(max_seq_items // 2, 10)
+                head = [formatter(x) for x in self[:n]]
+                tail = [formatter(x) for x in self[-n:]]
+                summary = '[{head} ... {tail}]'.format(
+                    head=', '.join(head), tail=', '.join(tail))
+            else:
+                head = []
+                tail = [formatter(x) for x in self]
+                summary = '[{tail}]'.format(tail=', '.join(tail))
+
```

Comment:
so there is lots of copying of other code because you can't share the Index base things for this. You need prob push some of this code (e.g. formatting) to something like IndexOpsMixin (or the like). and use that here.